src/Sarif/Visitors/InsertOptionalDataVisitor.cs

```diff
@@ -55,19 +55,7 @@ public override Run VisitRun(Run node)
                 }
             }
 
-            if (node.Results != null)
```

Comment:
>Results [](start = 21, length = 7)

results objects are mutated in place, so we don't need this. also by not calling the base visit, we weren't visiting other constructs (such as those that contain messages but aren;t result instances, like notifications)